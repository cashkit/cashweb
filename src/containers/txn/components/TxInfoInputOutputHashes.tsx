import React from 'react';
import { Transaction } from '../../../protos/bchrpc_pb';


export interface TxInfoInputOutputHashesProps {
  inputsList: Array<Transaction.Input.AsObject>,
  outputsList: Array<Transaction.Output.AsObject>,
  onClickHash: (txHash: Uint8Array | string) => void,
}


export const TxInfoInputOutputHashes = ({ inputsList, outputsList, onClickHash }
  : { 
    inputsList: Array<Transaction.Input.AsObject>,
    outputsList: Array<Transaction.Output.AsObject>,
    onClickHash: (txHash: Uint8Array | string) => void,
  }) => {
  let InputsComponent;
  if (inputsList){
  InputsComponent = inputsList.map((input) => {
      const addr = input.address
       return <div key={addr} className="content">bitcoincash:{addr}</div>
     })
  }
  let OutputsComponent;
  if (outputsList){
  OutputsComponent = outputsList.map((output) => {
       const addr = output.address
       return <div key={addr} className="content">bitcoincash:{addr}</div>
     })
  }
return(
  <>
    <div className="tile is-ancestor">
      <div className="tile is-parent">
       <article className="tile is-child box has-text-left">
         <p className="is-size-4 has-text-weight-medium">Inputs Address List ({inputsList?.length})</p>
         {InputsComponent}
       </article>
     </div>
      <div className="tile is-parent">
       <article className="tile is-child box has-text-left">
         <p className="is-size-4 has-text-weight-medium">Outputs Address List ({outputsList?.length})</p>
         {OutputsComponent}
       </article>
     </div>
      
    </div>
  </>
  )
}