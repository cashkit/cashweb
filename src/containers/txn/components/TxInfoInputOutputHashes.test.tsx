import React from 'react';
import renderer from 'react-test-renderer';
import { TxInfoInputOutputHashes, TxInfoInputOutputHashesProps } from './TxInfoInputOutputHashes';

const props: TxInfoInputOutputHashesProps = {
    inputsList: [],
    outputsList: [],
    onClickAddress: (txHash: Uint8Array | string) => {},
}

it('<TxInfoInputOutputHashes/> renders correctly', () => {
  const tree = renderer
    .create(<TxInfoInputOutputHashes {...props}></TxInfoInputOutputHashes>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});