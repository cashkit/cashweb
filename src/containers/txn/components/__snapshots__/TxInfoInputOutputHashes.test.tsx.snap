// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TxInfoInputOutputHashes/> renders correctly 1`] = `
<div
  className="tile is-ancestor"
>
  <div
    className="tile is-parent"
  >
    <article
      className="tile is-child box has-text-left"
    >
      <p
        className="is-size-4 has-text-weight-medium"
      >
        Inputs Address List (
        0
        )
      </p>
    </article>
  </div>
  <div
    className="tile is-parent"
  >
    <article
      className="tile is-child box has-text-left"
    >
      <p
        className="is-size-4 has-text-weight-medium"
      >
        Outputs Address List (
        0
        )
      </p>
    </article>
  </div>
</div>
`;
